transport/peer.go

```diff
@@ -0,0 +1,48 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peer.go -package=transporttest go.uber.org/yarpc/transport PeerIdentifier,Peer
+
+// PeerConnectionStatus maintains information about the Peer's connection state
+type PeerConnectionStatus int
+
+const (
+	// PeerAvailable indicates the Peer is available for requests
+	PeerAvailable PeerConnectionStatus = 1
```

Comment:
You should use [iota](https://golang.org/ref/spec#Iota) for this. It
translates to:

```
const (
  PeerAvailable PeerConnectionStatus = iota + 1
  PeerConnecting
  PeerUnavailable
)
```